sunpy/database/tables.py

```diff
@@ -340,6 +341,66 @@ def _from_query_result_block(cls, qr_block, default_waveunit=None):
             instrument=instrument, size=qr_block.size,
             wavemin=wavemin, wavemax=wavemax)
 
+    @classmethod
+    def _from_fido_search_result_block(cls, sr_block, default_waveunit=None):
```

Comment:
this method should be named "_from_dataretriever_response_block" Fido is the interface dataretriever is the subpackgae that handles the extra sources.